pkgs/development/mobile/androidenv/compose-android-packages.nix

```diff
@@ -254,6 +257,10 @@ rec {
       do
           ln -s $i $out/bin
       done
+
+      # create license files in top sdk path
+      mkdir $out/licenses
+      ${builtins.concatStringsSep "\n" (builtins.attrValues (builtins.mapAttrs (k: v: "echo -e '\n${v}' > $out/licenses/${k}") licenses))}
```

Comment:
I don't know what is your use case. I have spent some time this week playing with this PR trying to run a React Native project. In the end it turned out that for my project I didn't need any special license, it was enough with `config.android_sdk.accept_license = true`.

In any case, when using this branch I realized that the `licenses` folder is created in the root of the package. However, the SDK is inside `libexec/android-sdk` so there is no way to configure `$ANDROID_HOME` so both the licenses and the tools are found. I think that the `licenses` folder should be inside the SDK folder.

```diff
-      mkdir $out/licenses
-      ${builtins.concatStringsSep "\n" (builtins.attrValues (builtins.mapAttrs (k: v: "echo -e '\n${v}' > $out/licenses/${k}") licenses))}
+      ${builtins.concatStringsSep "\n" (builtins.attrValues (builtins.mapAttrs (k: v: "echo -e '\n${v}' > $out/libexec/sdk-manager/licenses/${k}") licenses))}
```

Reply:
@dicearr I didn't manage to work my react-native with that, but I found adding this license stuff just progressed me a little further so wanted to share. I didn't try accept_licnese though.

Would you mind sharing your setup for local dev / emulator and releasing process? Currently I'm using docker with openjdk to provide this environments as I didn't mange to do it in Nix, I emulator from nix though. 
If you have this working I would be happy to document it and put in NixOS docs as this stuff is not very well described.

Reply:
I have a `run-android.nix` file which is a combination of the [android documentation](https://github.com/NixOS/nixpkgs/blob/07c84f4de229cddd1abc87350bbe17a4725964b1/doc/languages-frameworks/android.section.md) with https://github.com/NixOS/nixpkgs/issues/72220#issuecomment-549609997. Executing `$ nix-shell run-android.nix` should make the application run in the device/emulator.

```nix
with import <nixpkgs> {
  config.android_sdk.accept_license = true;
};

let androidComposition = androidenv.composeAndroidPackages {
  toolsVersion = "25.2.5";
  platformVersions = [ "28" ];
};

in (buildFHSUserEnv {
  name = "react-native-env";
  targetPkgs = pkgs: ([
    androidComposition.androidsdk
    glibc
    openjdk
    nodejs-12_x
  ]);
  runScript = ''
    ;ANDROID_HOME=${androidComposition.androidsdk}/libexec/android-sdk \
      $(npm bin)/react-native run-android
  '';
})

.env
```

Reply:
I think this licenses enable user to execute `gradle` task (build-related) directly on a project.